### PR TITLE
[vcpkg_find_acquire_program] Add tool powershell

### LINKED ISSUE #15681: [vcpkg_copy_tool_dependencies][clang_cl_x64] Could not find PowerShell Core
**Host Environment**
- OS: Windows 10 - Visual Studio 2019 v16.8.4
- Compiler: clang_cl_x64

**To Reproduce**
Steps to reproduce the behavior, create a new project with the following vcpkg manifest:

vcpkg.json
```json
{
  "name": "program",
  "version-string": "0.0.1",
  "dependencies": [
    "boost-uuid",
    "boost-proto",
    "boost-log",
    "boost-filesystem",
    "boost-chrono",
    "protobuf",
    "grpc",
    "azure-storage-cpp",
    "cryptopp"
  ]
}
```

```Cmake
# CMakeList.txt : Top-level CMake project file, do global configuration
# and include sub-projects here.
#
cmake_minimum_required (VERSION 3.8)

project ("program")

# Include sub-projects.
add_subdirectory ("Processor")
add_subdirectory ("gRPC")
add_subdirectory ("internal_package ")
add_subdirectory ("Blob")
add_subdirectory ("Logging")

find_package(internal_package REQUIRED)
find_package(boost-uuid REQUIRED)
find_package(boost-proto REQUIRED)
find_package(boost-log REQUIRED)
find_package(boost-filesystem REQUIRED)
find_package(boost-chrono REQUIRED)
find_package(protobuf REQUIRED)
find_package(grpc REQUIRED)
find_package(azure-storage-cpp REQUIRED)
find_package(cryptopp REQUIRED)
```

**Failure logs** 
```
Severity	Code	Description	Project	File	Line	Suppression State
Error		CMake Error at scripts/cmake/vcpkg_copy_tool_dependencies.cmake:26 (message):
  Could not find PowerShell Core; please open an issue to report this.		C:\Users\SkywalkerIsNull\Repos\Debayer\src\Debayer\scripts/cmake/vcpkg_copy_tool_dependencies.cmake		
```
image

**Additional context**
Attempted to install all dependencies manually via powershell (via windows terminal and not in the VS IDE) and they all successfully compile and install.
Removed/uninstalled pwsh core via Windows Store
Modify the vcpkg/scripts/vcpkgTools.xml to from:
```xml
<tool name="powershell-core" os="windows">
```
to:
```xml
<tool name="pwsh" os="windows">
```

### PR DESCRIPTION
**Pull request by JackBoosY**
Add `powershell` to `vcpkg_find_acquire_program` to fix the issue that couldn't find powershell correctly.

Fixes #15681.

### CODE REVIEW

**Comment by JackBoosY** [comment]
Waiting for #15740.

**Comment by JackBoosY** [comment]
@skywalkerisnull  Can you please test this PR?

Thanks.

File: scripts/cmake/vcpkg_find_acquire_program.cmake [hunk 1]
```diff
@@ -115,6 +116,16 @@ function(vcpkg_find_acquire_program VAR)
       set(BREW_PACKAGE_NAME "git")
       set(APT_PACKAGE_NAME "git")
     endif()
+  elseif(VAR MATCHES "POWERSHELL")
+    set(PROGNAME pwsh)
+    if(CMAKE_HOST_WIN32)
+      set(PWSH_VERSION 7.1.10)
```
Comment (WangZishi) [inline scripts/cmake/vcpkg_find_acquire_program.cmake:122]: This PWSH_VERSION may be wrong. It should be 7.1.1.

Comment (JackBoosY) [reply scripts/cmake/vcpkg_find_acquire_program.cmake:122]: Should be 7.1.0

Comment (JackBoosY) [inline scripts/cmake/vcpkg_find_acquire_program.cmake:122]:
```suggestion
      set(PWSH_VERSION 7.1.0)
```

**Comment by JackBoosY** [comment]
Regressions will be fixed in #15973.

**Review by PhoebeHui — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Blocked on getting a response for https://github.com/microsoft/vcpkg/issues/15681#issuecomment-777684726

**Comment by PhoebeHui** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**